[nmstate-1.2] sriov: Only verify SRIOV VF when desired

cathay4t opened the pull request:
Skip the verification on SR-IOV if desire state does not mentioned so.

Unit test case included.

cathay4t committed:
sriov: Only verify SRIOV VF when desired

Skip the verification on SR-IOV if desire state does not mentioned so.

Unit test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (changes requested):
Please, fix the lint warnings.. other than that it looks good :)

cathay4t committed:
rust test: Skip test_linux_bridge_multicast_router

We have no intention to backport rust patch to 1.2 branch. Hence we skip
this test case.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
Fix rust clippy

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
ovs: Mark test_ovs_service_missing_with_system_port_only as optional to fail

The `test_ovs_service_missing_with_system_port_only` is OK to fail as it
is not a important use case.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
nm test: optional xfail test_linux_bridge_does_not_lose_unmanaged_port_on_rollback

The `test_linux_bridge_does_not_lose_unmanaged_port_on_rollback` is
fixed in NM 1.39.

Signed-off-by: Gris Ge <fge@redhat.com>
